logs.tf

```
@@ -35,67 +37,71 @@ resource "aws_athena_named_query" "waf_logs_table" {
 }
 
 resource "aws_athena_named_query" "requests_per_client_ip" {
-  count     = var.deploy_athena_queries ? 1 : 0
+  count     = var.deploy_logs_bucket ? 1 : 0
   name      = "requests-per-client-ip-per-5min-${var.waf_name}"
-  workgroup = aws_athena_workgroup.waf.id
-  database  = aws_athena_database.waf.name
+  workgroup = aws_athena_workgroup.waf[0].id
+  database  = aws_athena_database.waf[0].name
   query     = file("${path.module}/athena_queries/client_ip_per_5min.sql")
 }
 
 resource "aws_athena_named_query" "count_group_by" {
-  count     = var.deploy_athena_queries ? 1 : 0
+  count     = var.deploy_logs_bucket ? 1 : 0
   name      = "count-requests-grouped-by-ip-tenant-endpoint-${var.waf_name}"
-  workgroup = aws_athena_workgroup.waf.id
-  database  = aws_athena_database.waf.name
+  workgroup = aws_athena_workgroup.waf[0].id
+  database  = aws_athena_database.waf[0].name
   query     = file("${path.module}/athena_queries/count_requests_grouped_by_ip_tenant_endpoint.sql")
 }
 
 resource "aws_athena_named_query" "blocked_requests" {
-  count     = var.deploy_athena_queries ? 1 : 0
+  count     = var.deploy_logs_bucket ? 1 : 0
   name      = "requests-blocked-${var.waf_name}"
-  workgroup = aws_athena_workgroup.waf.id
-  database  = aws_athena_database.waf.name
+  workgroup = aws_athena_workgroup.waf[0].id
+  database  = aws_athena_database.waf[0].name
   query     = file("${path.module}/athena_queries/blocked_requests.sql")
 }
 
 resource "aws_athena_named_query" "per_ip_blocked_requests" {
-  count     = var.deploy_athena_queries ? 1 : 0
+  count     = var.deploy_logs_bucket ? 1 : 0
   name      = "requests-blocked-per-client-ip-${var.waf_name}"
-  workgroup = aws_athena_workgroup.waf.id
-  database  = aws_athena_database.waf.name
+  workgroup = aws_athena_workgroup.waf[0].id
+  database  = aws_athena_database.waf[0].name
   query     = file("${path.module}/athena_queries/per_ip_blocked_requests.sql")
 }
 
 resource "aws_athena_named_query" "first_logs_query" {
-  count     = var.deploy_athena_queries ? 1 : 0
+  count     = var.deploy_logs_bucket ? 1 : 0
   name      = "first-ten-results-${var.waf_name}"
-  workgroup = aws_athena_workgroup.waf.id
-  database  = aws_athena_database.waf.name
+  workgroup = aws_athena_workgroup.waf[0].id
+  database  = aws_athena_database.waf[0].name
   query     = "SELECT * FROM waf_logs limit 10;"
 }
 
 resource "aws_s3_bucket" "logs" {
+  count         = var.deploy_logs_bucket ? 1 : 0
   bucket        = coalesce(var.logs_bucket_name_override, "aws-waf-logs-${var.waf_name}-${data.aws_caller_identity.current.account_id}")
   force_destroy = true
 }
 
 # See issue <https://github.com/hashicorp/terraform-provider-aws/issues/28353>
 resource "aws_s3_bucket_ownership_controls" "logs" {
-  bucket = aws_s3_bucket.logs.id
+  count  = var.deploy_logs_bucket ? 1 : 0
+  bucket = var.deploy_logs_bucket ? aws_s3_bucket.logs[0].id : var.alternative_logs_bucket_name
```

why a condition for the bucket ? 
```suggestion
  bucket = aws_s3_bucket.logs[0].id
```

---

Because in case we have 2 wafs deployed we said that we would share the same bucket for collecting logs. This is why the condition and alternative log bucket, most likely used by the 2+ WAF.

---

Yes but if deploy_logs_bucket is false, you never go in this condition